### PR TITLE
Fix poetry dependency resolution in devcontainer and simplify setup

### PR DESCRIPTION
**Pull request by slangenbach**
**Description:** Fix poetry dependency resolution within devcontainer and simplify devcontainer configuration along the way:
- Specify devcontainer config in devcontainer.json only (get rid of dev Dockerfile)
- devcontainer uses same poetry version as container running unit tests
- Install poetry via devcontainer feature instead of using multi-stage build
- Move setup tasks (configuring shell completions, installing dependencies) to dedicated file (setup.sh)

**Issue:** 
- #7825 
- #8277 

**Dependencies:** None
**Tag maintainer:** @eyurtsev

### CODE REVIEW

**Comment by sdelgadoc** [comment]
Hey @slangenbach, thanks for working on this. I am getting the following error when I try to generate a devcontainer on GitHub Codespaces.

Will this PR fix the issue or should a create an issue too?

```
2330.6 Terminated
------
Dockerfile-with-features:46
--------------------
  44 |     
  45 |     # Install the Poetry dependencies (this layer will be cached as long as the dependencies don't change)
  46 | >>> RUN poetry install --no-interaction --no-ansi --with dev,test,docs
  47 |     
  48 |     FROM $_DEV_CONTAINERS_BASE_IMAGE AS dev_containers_target_stage
--------------------
ERROR: failed to solve: process "/bin/sh -c poetry install --no-interaction --no-ansi --with dev,test,docs" did not complete successfully: exit code: 143
ERROR: Service 'langchain' failed to build : Build failed
```

**Comment by slangenbach** [comment]
@sdelgadoc Yes with the caveat that my fix doesn't install dependencies for docs. If this is strictly required, I'll look into it. 

Docs' dependencies are listed in the pyproject.toml file at the root of the project. I focus on dependencies required for development specified in pyproject.toml at /libs/langchain

**Comment by sdelgadoc** [comment]
@slangenbach, thanks for all your work on this! I think that the dependencies you are focusing on should be fine. I'll test the devcontainer once it get merged into main, and let you know if I have any feedback here.

**Comment by slangenbach** [comment]
@nfcampos Is there anything else you need from my side in order to merge this PR? I guess you have many things on your plate, yet I'd really like to fix devcontainers for all of LangChain's contributors.

**Comment by sdelgadoc** [comment]
@nfcampos, I second @slangenbach's request. Currently, LangChain's devcontainers can not be generated, and this fix will resolve the issue.

**Comment by alan-copeland-keysight** [comment]
I found the devcontainer via your blog, and tried to use it yesterday but wasn't successful. So, there are definitely people who would benefit from this issue getting fixed.

**Comment by slangenbach** [comment]
@baskaryan Can you help out here?

**Comment by slangenbach** [comment]
@nfcampos Any thoughts on the PR?

**Comment by ZyeG** [comment]
@slangenbach Thank you so much for the work! It solves the issue for me. @nfcampos Can you please help with the merge? It would be a lot easier for future contributors.

**Comment by cgalo5758** [comment]
Confused as to why this PR was closed, the issue still persists.

**Comment by sansmoraxz** [comment]
@hwchase17 could you provide some reason on why this approach was rejected?

#15251 does not seem to work.

File: .devcontainer/setup.sh [hunk 1]
```diff
@@ -0,0 +1,5 @@
+#!/bin/bash
+poetry completions bash >> ~/.bash_completion
+cd libs/langchain
+# Installs main, test, codespell, lint, typing (all non-optional) and dev dependencies
+poetry install --no-interaction --no-ansi --with dev
```
Comment (cgalo5758) [inline .devcontainer/setup.sh:5]: Can you explain the change here? It used to be `--with dev,docs,test`

**Pull request closed without merging**